pkg/bmc/virtualmedia_validation.go

```diff
@@ -0,0 +1,56 @@
+package bmc
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// ValidateVirtualMedia returns nil when the BMC has support
+// for VirtualMedia and error when the BMC does not have support.
+func ValidateVirtualMedia(address string, user string, password string, insecure bool) error {
+	bmcConfig := gofish.ClientConfig{
+		Endpoint: address,
+		Username: user,
+		Password: password,
+		Insecure: insecure,
+	}
+
+	client, err := gofish.Connect(bmcConfig)
+	if err != nil {
+		return errors.Wrap(err, "could not connect to redfish BMC.")
+	}
+	defer client.Logout()
+
+	service := client.Service
+
+	managers, err := service.Managers()
+	if err != nil {
+		return errors.Wrap(err, "could not retrive redfish managers.")
+	}
+
+	for _, manager := range managers {
+		vmediaList, err := manager.VirtualMedia()
+		if err != nil {
+			return errors.Wrap(err, "could not access virtual media on manager")
+		}
+		if vmediaList == nil {
+			continue
+		}
+
+		for _, vmedia := range vmediaList {
+			for _, media := range vmedia.MediaTypes {
+				if media != redfish.CDMediaType {
+					continue
+				}
+
+				if vmedia.SupportsMediaInsert && vmedia.SupportsMediaEject {
+					return nil
+				}
+				return errors.New("missing one of the following Actions: InsertMedia, EjectMedia")
```

Comment:
nit: Add virtual media ID to the message

Reply:
ok

Reply:
Done